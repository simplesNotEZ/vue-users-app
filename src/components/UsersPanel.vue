<template>
  <div class="users-panel-container">
    <div class="addUserBtn-container">
      <button class="addUserBtn" @click="addUser">Add New User</button>
    </div>
    <h2>Current Users</h2>
    <User v-for="user in users" :user="user" :key="user.id" />
  </div>
</template>

<script>
import User from "./User.vue";

export default {
  name: "UsersPanel",
  components: {
    User
  },
  props: {
    users: Array
  },
  methods: {
    addUser() {
      this.$store.commit('updateFormStatus', 'addingUser');
      this.$router.push({ path: "/form" }); 
    }
  }
}
</script>

<style scoped>
.users-panel-container {
  display: flex;
  flex-direction: column;
  align-items: center;
}
.addUserBtn {
  border: 2px solid black;
  border-radius: .5rem;
  padding: .5rem;
  font-weight: bold;
  margin-top: .25rem;
  width: 16rem;
  cursor: pointer;
  text-transform: uppercase;
}
.addUserBtn:hover {
  background-color: #bfbfbf;
}
</style>
