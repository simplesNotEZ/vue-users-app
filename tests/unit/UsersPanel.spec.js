import { shallowMount } from "@vue/test-utils";
import UsersPanel from "@/components/UsersPanel.vue";

let wrapper;

beforeEach(() => {
  wrapper = shallowMount(UsersPanel, {
    propsData: {
      users: [  
        { 
          "id": 1,
          "name": "Leanne Graham",
          "username": "Bret",
          "email": "Sincere@april.biz",   
        },
        {
          "id": 2,
          "name": "Ervin Howell",
          "username": "Antonette",
          "email": "Shanna@melissa.tv",
        }
      ]
    }
  });
});

describe("UsersPanel", () => {
  it("renders the stubbed User child component(s)", () => {
    expect(wrapper.find('user-stub').exists()).toBe(true);
  });

  it("calls addUser method on click of Add User btn", async() => {
    const addUser = jest.fn();
    wrapper.setMethods({ addUser });
    await wrapper.find('button').trigger('click');
    expect(addUser).toHaveBeenCalled();
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
