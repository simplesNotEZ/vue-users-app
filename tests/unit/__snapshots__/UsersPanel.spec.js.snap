// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersPanel renders correctly 1`] = `
<div class="users-panel-container">
  <div class="addUserBtn-container"><button class="addUserBtn">Add User</button></div>
  <h2>Current Users</h2>
  <user-stub user="[object Object]"></user-stub>
  <user-stub user="[object Object]"></user-stub>
</div>
`;
